src/main/java/com/capstone/doctorsOffice/controllers/PatientController.java

```diff
@@ -0,0 +1,43 @@
+package com.capstone.doctorsOffice.controllers;
+
+import com.capstone.doctorsOffice.dtos.PatientDto;
+import com.capstone.doctorsOffice.repositories.PatientRepository;
+import com.capstone.doctorsOffice.services.PatientService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/patient")
+public class PatientController {
+
+    @Autowired
+    private PatientService patientService;
+    @Autowired
+    private PatientRepository patientRepository;
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+
+    @PostMapping("/register")
+    public List<String> addPatient(@RequestBody PatientDto patientDto){
+        String passHash = passwordEncoder.encode(patientDto.getPassword());
+        patientDto.setPassword(passHash);
+        return patientService.addPatient(patientDto);
+    }
+
+    @PostMapping("/login")
+    public List<String> patientLogin(@RequestBody PatientDto patientDto){
+        return patientService.patientLogin(patientDto);
+    }
+
+    @GetMapping("/{id}")
+    public Optional<PatientDto> getPatientById(@PathVariable Long patientId) {
+        return patientService.getPatientById(patientId);
```

Comment:
the variable name needs to be the same as what is in the path.
```
@GetMapping("/{id}")
     public Optional<PatientDto> getPatientById(@PathVariable Long id)
```

If you would like to use a different name for the variable you would have to specify the PathVariable name by passing it as a parameter
Example:
```
@GetMapping("/{id}")
     public Optional<PatientDto> getPatientById(@PathVariable("id") Long patientId)
```

Reply:
Done